railties/lib/rails/generators/rails/authentication/templates/app/controllers/passwords_controller.rb.tt

```diff
@@ -7,11 +7,13 @@ class PasswordsController < ApplicationController
   end
 
   def create
+  <% if defined?(ActionMailer::Railtie) -%>
```

Comment:
Shouldn't this action just not exist if ActionMailer is skipped? It's basically a noop.

The route would probably need to be removed to?

Reply:
Ah yes, of course! I'll take a crack at adjusting the routes and removing `#create` and `#new`.

One thing I noticed is the way the routes are set up would also generate `#show`, `#index`, and `#destory`. Would it make sense to always have an `only:` in there? Then if the mailer is defined it would be `only: [:new, :create, :edit, :update]`, otherwise `only: [:edit, :update]`

https://github.com/rails/rails/blob/7cb69a8fd004c94b9266f2fdb2a833d15c09192b/railties/lib/rails/generators/rails/authentication/authentication_generator.rb#L40-L43

Reply:
I think adding `only:` to generated routes would not be accepted: https://github.com/rails/rails/pull/52369#discussion_r1699079086

Reply:
@skipkayhil Thank you for the heads up! 🙏 

@byroot I think I addressed the feedback fully. Let me know if you think any more changes would be appropriate here.